libs/core/form/form-control/form-control.component.ts

```diff
@@ -92,15 +183,38 @@ export class FormControlComponent implements CssClassBuilder, OnInit, OnChanges,
     buildComponentCssClass(): string[] {
         const tagName = this.elementRef.nativeElement.tagName.toLowerCase();
         return [
-            this.state ? 'is-' + this.state : '',
-            this.class,
-            tagName === 'textarea' ? 'fd-textarea' : tagName === 'input' ? 'fd-input' : ''
-        ];
+            tagName === 'textarea' ? 'fd-textarea' : tagName === 'input' ? 'fd-input' : '',
+            this.state() ? `is-${this.state()}` : '',
+            this.class()
+        ].filter(Boolean) as string[];
```

Comment:
what is this line doing?

Reply:
we had this:
```
[
  tagName === 'textarea' ? 'fd-textarea' : tagName === 'input' ? 'fd-input' : '',
  this.state() ? `is-${this.state()}` : '',
  this.class()
]
```
if a certain condition doesn't match, the fallback is an empty string ''. `.filter(Boolean)` will remove empty strings, null, undefined, false, 0. In this case, it mainly strips out the '' from when no class applies. Since Boolean isn’t a strict type guard we can get `(string | undefined | null)[]`, that's why we need to make sure we get an array of strings with `as string[]`.

Example:
```
['fd-input', '', 'extra-class'].filter(Boolean)
```

will result in:
```
['fd-input', 'extra-class']
```